core/core-impl/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/MoveSelectorConfig.java

```diff
@@ -280,4 +282,33 @@ private void inheritCommon(MoveSelectorConfig<?> inheritedConfig) {
                 fixedProbabilityWeight, inheritedConfig.getFixedProbabilityWeight());
     }
 
+    protected String addRandomSuffix(String name, Random random) {
+        StringBuilder value = new StringBuilder(name);
+        value.append("-");
+        random.ints(97, 122) // ['a', 'z']
+                .limit(4) // 4 letters
+                .forEach(value::appendCodePoint);
+        return value.toString();
+    }
+
+    /**
+     * Check if the selector factory accepts Nearby selection autoconfiguration.
+     *
+     * @return false by default
+     */
+    public boolean acceptNearbySelectionAutoConfiguration() {
+        return false;
+    }
```

Comment:
I'd consider working without this. You can call `enableNearbySelection(...)`, which by default returns `null`. If null, nothing happens.

(I'm driven by eliminating boilerplate code; the selectors already have plenty without adding more.)

Reply:
We could (maybe) even go as far as to create a new sub-class for nearby-selection enabled selectors, so that it is absolutely obvious on a type level whether or not something supports nearby selection.

Reply:
I liked the second approach and will proceed with implementing it.